generic-server/src/main.rs

```diff
@@ -1,28 +1,43 @@
-use generic_server::app;
-
 use axum::Server;
+use generic_server::app;
 use std::net::SocketAddr;
 
 #[tokio::main]
 async fn main() {
+    // Start server
+    let port = std::env::var("SERVER_LOCAL_PORT").unwrap_or("3000".to_owned());
+    let addr: SocketAddr = format!("127.0.0.1:{port}").parse().unwrap();
+    tracing::info!("listening on {addr}");
+    generic_server_with_gracefull_shutdown(addr).await
+}
+
+async fn generic_server_with_gracefull_shutdown(addr: SocketAddr) {
     // Load dotenv-flow variables
     dotenv_flow::dotenv_flow().ok();
 
     // Enable logging
     config_tracing();
 
-    // Start server
-    let port = std::env::var("SERVER_LOCAL_PORT").unwrap_or("3000".to_owned());
-    let addr: SocketAddr = format!("127.0.0.1:{port}").parse().unwrap();
-    tracing::info!("listening on {addr}");
-    Server::bind(&addr)
-        .serve(app().into_make_service())
-        .await
-        .unwrap();
-}
+    // Load plugins
+    let (mut plugin_container, router) = app();
 
+    // spawn task for server
+    tokio::spawn(async move {
+        Server::bind(&addr)
+            .serve(router.into_make_service())
+            .await
+            .unwrap();
+    });
+
+    tokio::select! {
+        _ = tokio::signal::ctrl_c() => {
+           tracing::info!("\nshuting down gracefully");
```

Comment:
```suggestion
           tracing::info!("shutting down gracefully");
```

Reply:
> 

Hey @IngridPuppet I don't see the problem of logging the unmounted plugins on a new line

Reply:
The logs can be parsed by another program. You are making difficult with the new line.